ZenPacks/zenoss/LinuxMonitor/FileSystem.py

```diff
@@ -0,0 +1,31 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2016, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+from Products.ZenModel.FileSystem import FileSystem as BaseFileSystem
+from Products.Zuul.interfaces import ICatalogTool
+
+
+class FileSystem(BaseFileSystem):
+    logicalvolume = None
+    meta_type = 'LinuxFileSystem'
+
+    _properties = BaseFileSystem._properties + (
+        {'id': 'logicalvolume', 'label': 'Logical Volume',
+            'type': 'string', 'mode': 'w'},
+        )
+
+    def getLogicalVolume(self):
+        results = ICatalogTool(self.device()).search(
```

Comment:
Using the global catalog to search locally like this is expensive. We can use zenpacklib's automatic indexing to optimize this by using a device-local catalog, and searching specifically for LVs and snapshots with the right mountpoint value.

First add indexing to the mountpoint property on LogicalVolume and SnapshotVolume in zenpack.yaml

```yaml
  properties:
    mountpoint:
      index_type: field
```

Then the implementation of FileSystem.getLogicalVolume() would look like this..

```python
import itertools
from . import zenpacklib

class FileSystem(BaseFileSystem):
    def getLogicalVolume(self):
        results = itertools.chain.from_iterable(
            zenpacklib.catalog_search(
                self.device(),
                name,
                mountpoint=self.mount)
            for name in ('LogicalVolume', 'SnapshotVolume'))

        for result in results:
            try:
                return result.getObject()
            except Exception:
                pass
```